src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCertificateUtils.java

```diff
@@ -153,22 +154,23 @@ private static PrivateKey loadPrivateKeyFromPVK(String keyPath,
         ByteBuffer buffer = ByteBuffer.allocate((int) f.length());
         try (FileInputStream in = new FileInputStream(f)) {
             in.getChannel().read(buffer);
-            buffer.order(ByteOrder.LITTLE_ENDIAN).rewind();
+            ((Buffer) buffer.order(ByteOrder.LITTLE_ENDIAN)).rewind();

```

Comment:
Is this really necessary, the [Oracle Java docs](https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html) says rewind is inherited from Buffer since Java SE7.

Reply:
I'm curious about this one as well. Does Java try to use the Java 9 version of rewind from ByteBuffer without this cast?

Reply:
yes that is the reason, ByteBuffer doesn't have this method so it will throw a NoSuchMethodError